### ISSUE TITLE
Rendering and Physics mirrored axis

### ISSUE THREAD

**Issue by FaicSilier** [x3dom/x3dom#604]
Hello,
When I load a non symmetric object with indexedFaceSet to display it and load the same indexedFaceSet as collidableShape in the physic, it is mirrored in the physic.

Unfortunately there is no way to scale the physic model of my object by -1 because there is no way to scale the physic model at all (at least I haven't found a way).

**Comment by andreasplesch** [comment]
An example scene would probably help to understand and track down the issue.

**Comment by FaicSilier** [comment]
I made an example szene with 1.7.1

you have to rename the txt to html. My example scene is based on the physic example on the x3dom website.
[test.txt](https://github.com/x3dom/x3dom/files/113404/test.txt)

**Comment by andreasplesch** [comment]
Thanks. I copied the example to a gist here:

https://gist.github.com/andreasplesch/0a7baab98866d3ba478d

It is served as html via rawgit here:

https://rawgit.com/andreasplesch/0a7baab98866d3ba478d/raw/30f7cd88c51b9ae4b8f1717ec093c78959502052/FaicSillier_mirroredPhysics.html

That is an easy way to share such scenes.  Do you want to do this on your github account ?

Looks cool. What is the small red car supposed to do ? Is it colliding with a mirrored (Y=-Y?) ground surface rather than the actual ground surface ?

**Comment by FaicSilier** [comment]
The red car is from the example from the x3dom website. I just exchanged the floor with a indexedFaceSet I had from another project. (A box would also do the job)

But as you can see, the car is colliding with a mirrored surface even if the indexedFaceSets are exactly the same. I think you have to mirror X(-X) and Y(-Y) to make both surfaces match. So here is a bug/error in the physics.

**Comment by andreasplesch** [comment]
It may be necessary to boil the scene down to a minimal set of boxes which show the problem since the problem does not affect the example scenes.
I noticed that the floor in your scene defines explicit normals whereas in the examples the green plane does not seem to have normals. Did you try the scene without explicit normals ?

**Comment by FaicSilier** [comment]
I reduced the scene to a minimum. Included is also the correct mirrored version of the indexedFacedSet ... wich is mirrored by x=-x and y=-y

https://gist.github.com/FaicSilier/0f92addd6e59b2b08d90

**Comment by andreasplesch** [comment]
live
http://rawgit.com/FaicSilier/0f92addd6e59b2b08d90/raw/edc6c089490b75b5002d02d580f04d39037f2038/physics.html

**Comment by andreasplesch** [comment]
The scene should be helpful. The red boxes fall through the green flat slab at the bottom. Can you confirm this is intentional ?

Some of the red boxes collide with a corner of the large blue box. The large blue box needs to be mirrored as collidable shape for physics purposes to correspond to the rendered box. Then the red boxes collide correctly. Can you confirm that this is the idea ?

The mirroring seems to be only required for fixed='true' rigid bodies ? Since the red boxes have physics corresponding to their rendered positions ?

The other thing to figure out would be is why the ground surface in the examples does not need to be mirrored ? Is it symmetrical in the first place ?

**Comment by FaicSilier** [comment]
yes, the green floor is just for orientation purpose. Collision only happens between the red boxes and blue box.

Yes, if you copy the indexedFaceSet to be used with the physics the displayed version and the physics version don't match. You have to mirror the physics indexedFaceSet with x=-x and y=-y to match both objects.

The red boxes are point symmetric so they match their physics model. The bug is independent of the "fixed" attribute. If you set fixed to false and give it a mass of 20000 so that it won't move the error is still there.

Yes, everything which is point symmetric is not affected by this error.

**Comment by andreasplesch** [comment]
Just to clarify one more bit. It looks like the point about which objects need to be point symmetric to hide this issue is the local origin (0,0,0) of the geometry of the shape (before any transformation). In contrast the big blue box is also point symmetric but about a point offset from the origin in local coordinates, and therefore has the problem. Is this your understanding as well ?

This is probably as far as I can help to pin point the issue. To me it looks like ammo.js (which is used for physics) may have a convention for the coordinate system which is different from x3d's coordinate system and which is not fully accounted for in how x3dom uses ammo.js . It would be best if someone who has experience with the rigid body physics component could take a look.

Is one of the scene here
http://www.web3d.org/x3d/content/examples/Basic/RigidBodyPhysics/
also affected ?
(web3d does not link in the correct x3dom-full-physics.js)

**Comment by FaicSilier** [comment]
Yes, point symmetric is only "working" if the origin is at (0,0,0).

Same for me, I'm not using physics anymore so I will not put more time in it to fix the error. But I think we pin pointed the issue very precisely.

**Comment by Dodo-aa** [comment]
Hi, I don't know how to say that but for me, it was really hard to understand physics and chemistry if there is any idea to help me I would be happy, and thank you 🤝
